File: java/program79.java
import java.util.*;

class ArrayX
{
    public int Arr[];

    ArrayX(int iSize)
    {
        this.Arr = new int[iSize];
    }

    public void Accept()
    {
        System.out.println("Enter elements of the array : ");

        Scanner sobj = new Scanner(System.in);
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("Elements of the array are : ");
        int iCnt = 0;
        
        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.print(Arr[iCnt]+"\t");
        }
        System.out.println();
    }
}

class MyArray extends ArrayX
{
    MyArray(int size)
    {
        super(size);
    }

    public int Addition()
    {
        int sum = 0;
        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            sum = sum + Arr[iCnt];
        }
        return sum;
    }
}

class program79
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number elements");
        int size = sobj.nextInt();

        MyArray obj = new MyArray(size);
        obj.Accept();
        obj.Display();
        
        System.out.println("Addition is : "+obj.Addition());
    }
}

File: cpp/program79.cpp
// program76.c

#include<iostream>
using namespace std;

class ArrayX
{
    public:
        int *Arr;
        int iSize;

        ArrayX(int);
        ~ArrayX();
        
        void Accept();
        void Display();
        int Addition();
};

ArrayX::ArrayX(int no)
{
    iSize = no;
    Arr = new int[iSize];
}

ArrayX::~ArrayX()
{
    delete []Arr;
}

void ArrayX::Accept()
{
    cout<<"Enter elements of array \n";

    for(int iCnt = 0; iCnt < iSize; iCnt++)
    {
        cin>>Arr[iCnt];
    }
}

void ArrayX::Display()
{
    cout<<"Elements of the array are \n";
    for(int icnt = 0; icnt < iSize; icnt++)
    {
        cout<<Arr[icnt]<<"\t";
    }
    cout<<"\n";
}

int ArrayX::Addition()
{
    int iSum = 0;

    for(int iCnt = 0; iCnt < iSize; iCnt++)
    {
        iSum = iSum + Arr[iCnt];
    }
    return iSum;
}

int main()
{   
    int iSize = 0;

    cout<<"Enter number of elements \n";
    cin>>iSize;

    ArrayX obj(iSize);
    obj.Accept();
    obj.Display();

    cout<<"Addition of the array elements is : "<<obj.Addition()<<"\n";

    return 0;
}